test/Concurrency/actor_isolation.swift

```diff
@@ -0,0 +1,109 @@
+
+func globalFunc() { }
+func acceptClosure<T>(_: () -> T) { }
+func acceptEscapingClosure<T>(_: @escaping () -> T) { }
+
+// ----------------------------------------------------------------------
+// Actor state isolation restrictions
+// ----------------------------------------------------------------------
+actor class MySuperActor {
+  var superState: Int = 25
+
+  func superMethod() { }
+  func superAsyncMethod() async { }
+
+  subscript (index: Int) -> String { // expected-note{{subscript declared here}}
+    "\(index)"
+  }
+}
+
+actor class MyActor: MySuperActor {
+  let immutable: Int = 17
+  var text: [String] = [] // expected-note 4{{mutable state is only available within the actor instance}}
+
+  class func synchronousClass() { }
+  static func synchronousStatic() { }
+
+  func synchronous() -> String { text.first ?? "nothing" } // expected-note 3{{only asynchronous methods can be used outside the actor instance; do you want to add 'async'?}}
+  func asynchronous() async -> String { synchronous() }
+}
+
+extension MyActor {
+  func testAsynchronous(otherActor: MyActor) async {
+    _ = immutable
+    _ = synchronous()
+    _ = text[0]
+
+    // Accesses on 'self' are okay.
+    _ = self.immutable
+    _ = self.synchronous()
+    _ = await self.asynchronous()
+    _ = self.text[0]
+    _ = self[0]
+
+    // Accesses on 'super' are okay.
+    _ = super.superState
+    super.superMethod()
+    await super.superAsyncMethod()
+    _ = super[0]
+
+    // Accesses on other actors can only reference immutable data or
+    // call asychronous methods
+    _ = otherActor.immutable // okay
+    _ = otherActor.synchronous() // expected-error{{actor-isolated instance method 'synchronous()' can only be referenced on 'self'}}
+    _ = await otherActor.asynchronous()
+    _ = otherActor.text[0] // expected-error{{actor-isolated property 'text' can only be referenced on 'self'}}
+
+    // Global data is okay if it is immutable.
+    _ = immutableGlobal
+    _ = mutableGlobal // expected-warning{{reference to var 'mutableGlobal' is not concurrency-safe because it involves shared mutable state}}
+
+    // Global functions are not actually safe, but we allow them for now.
+    globalFunc()
+
+    // Class methods are okay.
+    Self.synchronousClass()
+    Self.synchronousStatic()
+
+    // Closures.
+    let localConstant = 17
+    var localVar = 17 // expected-note 2{{var declared here}}
+    acceptClosure {
+      _ = text[0] // expected-warning{{actor-isolated property 'text' is unsafe to reference in code that may execute concurrently}}
+      _ = self.synchronous() // expected-warning{{actor-isolated instance method 'synchronous()' is unsafe to reference in code that may execute concurrently}}
+      _ = localVar // expected-warning{{local var 'localVar' is unsafe to reference in code that may execute concurrently}}
+      _ = localConstant
+    }
+
+    acceptEscapingClosure {
```

Comment:
It's strange that `self.synchronous` is left out of the `acceptEscapingClosure` cases.

Reply:
Error of omission, I'll fix before merging. Thanks!